cmd/registry-replacer/main.go

```diff
@@ -600,6 +635,88 @@ func pruneReplacements(config *api.ReleaseBuildConfiguration, filter asDirective
 	return utilerrors.NewAggregate(errs)
 }
 
+// when pruning unused base images we'll check the fully-resolved config to make sure an image is not used directly in
+// the config, or within any of the tests.
+func pruneUnusedBaseImages(config *api.ReleaseBuildConfiguration, resolvedConfig *api.ReleaseBuildConfiguration) {
+	usedBaseImages := sets.NewString()
+	for _, step := range resolvedConfig.RawSteps {
```

Comment:
What about
* PipelineImageCacheStepConfiguration
* SourceStepConfiguration
* ProjectDirectoryImageBuildStepConfiguration
* RPMServeStepConfiguration
* OutputImageTagStepConfiguration
?

/cc @bbguimaraes 
As I am not 100% familiar with the intrinsics of all steps types either

Reply:
I looked through all of the steps and did not think that any of those steps would need to be taken into consideration here, but I could definitely be wrong.

Reply:
Ok, this is something that _looks_ to me like it is needed but I am not really sure, lets wait for @bbguimaraes or @stevekuznetsov to chime in

Reply:
We'd definitely have to look at all steps that can reference one of the base images, since not doing so and removing the images would result in an invalid configuration.  Some of those steps are even in use in `openshift/release`:

```
$ jq -r < config.json '[.[].raw_steps[]?|keys]|flatten|sort|unique[]'
output_image_tag_step
pipeline_image_cache_step
project_directory_image_build_step
rpm_image_injection_step
```

The annoyance of having to reinvent code to do this every time is what I was complaining about in the context of `DPTP-2259`.  There is no good way to enumerate these steps statically, but any one whose `Requires` method returns an `InternalImageLink` with a reference to the `pipeline` stream qualifies.

Reply:
Ah ok, thanks for letting me know! I did run the `configresolver` after the changes and all of the configs in `openshift/release` did pass, so I assumed the changes were legit. I'll go back and ensure that all of those other steps are taken into account.

Reply:
I added some additional logic to account for the other step configurations. Hopefully this takes care of it, though I'm still a bit fuzzy on all of the various step types, so let me know if I'm still missing something.